### ISSUE TITLE
boolean-prop-naming does not check Flow props

### ISSUE THREAD

**Issue by avaly** [yannickcr/eslint-plugin-react#1540]
The `boolean-prop-naming` does not complain about boolean props defined in Flow which do not match the rule pattern.

**Comment by jomasti** [comment]
Can you provide an example? I took a quick look at the tests for `boolean-prop-naming`, and I see some that use props defined in the Flow style. The rule also appears to be checking type annotations. Maybe it is just missing your specific use case.

**Comment by Elindorath** [comment]
I ran into the same issue. This test should fail but it doesn't:
```javascript
{
  // ES6 components and Flowtype booleans
  code: `
    type Props = {
      something: boolean,
    };

    class Hello extends React.Component<Props> {
      render () { return <div />; }
    }
  `,
  options: [{
    rule: '^is[A-Z]([A-Za-z0-9]?)+'
  }],
  parser: parsers.BABEL_ESLINT
}
```

**Comment by kddial** [comment]
I also experience this bug with the conditions of:
- using a React Class component
- using Flow types

Here is an example component that does not throw boolean-prop-naming error but should
```
// @flow
import React from "react";

type Props = {
  disabled: boolean,
};
class TestComponent extends React.Component<Props> {
  render() {
    const { disabled } = this.props;
    return <div>{disabled ? "is disabled" : "not disabled"}</div>;
  }
}

export default TestComponent;
```

**Comment by EvHaus** [comment]
This issue seems to have fixed itself for me after migrating to `@babel/eslint-parser`. However, I noticed that the rule doesn't play nicely with the `react/prefer-read-only-props` which adds a `+` to every prop. I end up seeing:

```
Prop name (+) doesn't match rule (^(is|has)[A-Z]([A-Za-z0-9]?)+)
```
